Framework/DataHandling/inc/MantidDataHandling/PulseIndexer.h

```diff
@@ -35,36 +35,54 @@ class MANTID_DATAHANDLING_DLL PulseIndexer {
   PulseIndexer(std::shared_ptr<std::vector<uint64_t>> event_index, const std::size_t firstEventIndex,
```

Comment:
in the docstring above, could you write a bit about the physical origin of `firstEventIndex`? Functions like `getStartEventIndex` and `getStopEventIndex` shift the event index by this quantity, so it seems there are two event indexes at play here.